### DISCUSSION TITLE
MultiFab* levelbcdata in "setLevelBC" function when using Robin BC

### DISCUSSION THREAD

**Discussion by ZiyangHuang** [AMReX-Codes/amrex#3373] [Q&A] [answered]
Hi,

I am trying to solve a linear system with Robin BC. From the definition of function "setLevelBC":

virtual void setLevelBC (int amrlev, const MultiFab* levelbcdata,
                                      const MultiFab* robinbc_a = nullptr,
                                      const MultiFab* robinbc_b = nullptr,
                                      const MultiFab* robinbc_f = nullptr) = 0;

the Robin BC is well defined with input "robinbc_a", "robinbc_b", and "robinbc_f". I am not sure what data to input to "levebcdata".

Thanks!

**Comment by WeiqunZhang** [comment, marked as answer]
If all BCs are Robin, you can use `nullptr` as `levelbcdata`.

**Reply by ZiyangHuang** [reply]
Thanks a lot. I think "nullptr" can also be used when all BC's are periodic, correct?

**Reply by WeiqunZhang** [reply]
Right.
